98_exercises.Rmd

```diff
@@ -0,0 +1,78 @@
+---
+title: "Exercises"
+output: html_notebook
+---
+
+Here you can find assignments on different topics. 
+
+TIPS:
+
+   - Create a reproducible workflow using Rmd file. 
+   - Add comments that explain what each line or lines of code do. This helps you and others to understand your code and find bugs. Furthermore, it is easier for you to reuse the code, and it promotes transparency.
+   - Interpret results by comparing them to literature. List main findings, so that results can easily be understood by others without advanced data analytics knowledge.
+   - Avoid hard-coding. Use variables which get values in the beginning of the pipeline. That way it is easier for you to modify parameters and reuse the code.
+
+## Introduction to TreeSE
+
+Import data from CSV files to TreeSE.
+
+1. Import data
+2. Construct a TreeSE.
+3. Check that importing is done correctly. E.g., you can choose random samples and features,
+and check that their values equal between raw files and TreeSE.
+
+Functions that might be useful: DataFrame, TreeSummarizedExperiment, matrix, rownames, colnames, SimpleList
+
+## Data exploration
+
+1. Please install the latest development version of mia from GitHub.
+2. Load experimental dataset from mia.
+3. What are the dimensions? (How many samples there are, and how many taxa in each taxonomic rank)?
+4. Calculate library size.
+5. Summarize sample metadata variables. (How many age groups, how they are distributed? 0%, 25%, 50%, 75%, and 100% quantiles of library size?)
+6. Create two histograms. Another shows the distribution of absolute counts, another shows how CLR transformed values are distributed.
+7. Visualize how relative abundances are distributed between taxa in samples.
+8. Calculate Shannon diversity index. Is it significantly different between vegan and mixed diet?
+
+Functions that might be useful: nrow, ncol, dim, summary, table, quantile, unique, transformSamples, ggplot, wilcox.test, addPerCellQC, agglomerateByRank, estimateDiversity, plotAbundance
+
```

Comment:
Suggestions if situation allows. Here and in other tasks.

- add hyperlinks to function manpages.. or highlight the code
- add cross-links to relevant chapters in OMA